x-pack/filebeat/module/google_workspace/config/common.js

```diff
@@ -25,15 +25,16 @@ var googleWorkspace = (function () {
             "json.id.applicationName",
             "json.id.customerId",
         ];
+        var dynKeyArr = [];
```

Comment:
This whole section looks like it is working around limitations in the beat fingerprint processor. And it's not entirely overcoming those limitations (hashing arrays and objects). I think the would be reasonable to enhance the fingerprint processor to accommodate this usage (the ES fingerprint processor does this).

But I would consider fingerprinting on the JSON string that is contained in `message` if the full contents are suitable for a fingerprint (i.e. each time an event is requested the api returns the same JSON).

Reply:
I'm a bit worried atm to change the fingerprint processor behaviour since changing it there would impact all modules utilising the processor and could possibly cause an initial wave of duplicates all across the board since the sorting will impact the fingerprint. I'll bring it up in the team sync once to discuss this and then make the change. For now I'm merging this change(otherwise it will go past FF) and we'll see and observe if this impacts duplicate creation in any way.